os/binfmt/binfmt_execmodule.c

```diff
@@ -217,14 +217,12 @@ int exec_module(FAR struct binary_s *binp)
 	 * for calculating the heap usage per threads.
 	 */
 	int hashpid = PIDHASH(getpid());
-	struct mm_allocnode_s *node;
 
 	binp->uheap->alloc_list[hashpid].pid = HEAPINFO_INIT_INFO;
 	binp->uheap->alloc_list[hashpid].curr_alloc_size = 0;
 	binp->uheap->alloc_list[hashpid].num_alloc_free = 0;
 
-	node = (struct mm_allocnode_s *)((void *)stack - SIZEOF_MM_ALLOCNODE);
-	node->pid = (-1) * (newtcb->cmn.pid);
+	heapinfo_update_stack_pid(stack, newtcb->cmn.pid);
```

Comment:
Heapinfo calculation for stack node is in up createstack. But setting stack node is here. How about doing them at the same location? I don't mean the same function.

Reply:
I don't want to adjust the location of `task_schedsetup`. How about moving heapinfo code which is in `up_createstack` to here?

Reply:
yes, applied, Thanks~